pkg/ccl/streamingccl/streamproducer/span_config_event_stream.go

```diff
@@ -237,16 +243,45 @@ func (s *spanConfigEventStream) streamLoop(ctx context.Context) error {
 			}
 			batcher.addSpanConfigs(bufferedEvents)
 			bufferedEvents = bufferedEvents[:0]
-			if pacer.shouldCheckpoint(update.Timestamp, true) {
+			// Why if I gate the flushing with length of the span configs in the batch, no span config updates ever surface?
```

Comment:
If you have any ideas, I'm all ears. Even after running this through a debugger dozens of times, I couldn't figure it out. Happy to chat through the problem over a call. 

I suspect it has something to do with the fact that flushEvent flushes things to a channel, and the client polls that channel.

Reply:
Hold off before you spend time on this mystery. I thought of another debugging approach.